drivers/clk/renesas/rzg2l-cpg.c

```diff
@@ -158,6 +164,47 @@ struct rzg2l_cpg_priv {
 	struct rzg2l_pll5_mux_dsi_div_param mux_dsi_div_params;
 };
 
+/**
+ * get_dsi_status - Retrieves the status of the DSI node from the device tree.
+ * @dev: Pointer to the device structure.
+ *
+ * This function searches the device tree for the DSI node identified by the
+ * compatible string "renesas,r9a07g044-mipi-dsi" and reads its "status"
+ * property. It determines if the DSI node is enabled or disabled.
+ *
+ * Return: 1 if the DSI node is enabled, 0 if it is disabled,
+ *         -ENODEV if the DSI node is not found,
+ *         -EINVAL if the "status" property is missing.
+ */
+static int get_dsi_status(void)
+{
+    struct device_node *dsi_node;
+    const char *status;
+
+    /* Get the DSI node from the device tree */
+    dsi_node = of_find_compatible_node(NULL, NULL, "renesas,r9a07g044-mipi-dsi");
+    if (!dsi_node) {
+        pr_err("DSI node not found\n");
+        return -ENODEV;
+    }
+
+    /* Read the status property from the DSI node */
+    status = of_get_property(dsi_node, "status", NULL);
+    if (!status) {
+        pr_err("DSI node does not have a status property\n");
+        return -EINVAL;
+    }
+
+    /* Check if the DSI node status is "disabled" */
+    if (strcmp(status, "disabled") == 0) {
+        printk("DSI is disabled\n");
+        return 0;
+    } else {
+        printk("DSI is enabled\n");
+        return 1;
+    }
+}
+
```

Comment:
I don't think we should keep this change. Using the device tree to get the status of a specific compatible node is not ideal. We need a better approach.

Refer to: https://github.com/Renesas-SST/linux-rz/blob/dunfell/rz-sbc/drivers/gpu/drm/rcar-du/rcar_du_crtc.c#L315

Reply:
The reference code is inside encoder-bridge chain so they can easily get the status throughout the chain. But in this code we are in clock setting driver, so I don't see any way to get information from that chain without refer to a device node

Reply:
So, the function needs to be generic to support different device nodes and avoid hardcoding a specific node.

Reply:
I think using `of_find_node_by_name(NULL, "dsi");` could be more generic compared to `of_find_compatible_node(NULL, NULL, "renesas,r9a07g044-mipi-dsi");`, because ultimately, we are using this function to support DSI mode only. Therefore, letting the device search for the DSI node should be fine.

Reply:
But you know the DSI node name could be present under other compatible nodes, for example:
```
renesas,r9a07g054-mipi-dsi
```
so, we should modify the function to pass the compatible node as a parameter to get_dsi_status.